spec.html

```diff
@@ -7875,6 +7880,181 @@ <h1>Forward Progress</h1>
+        </p>
+
+        <p>
+          Because calling HostEnqueueFinalizationRegistryCleanupJob is optional, registered objects in a FinalizationRegistry do not necessarily hold that FinalizationRegistry live. Implementations may omit FinalizationRegistry callbacks for any reason, e.g., if the FinalizationRegistry itself becomes dead, or if the application is shutting down.
+        </p>
+      </emu-note>
+      <emu-note>
+        <p>
+          Implementations are not obligated to empty WeakRefs for maximal sets of non-live objects.
+        </p>
+        <p>
+          If an implementation chooses a non-live set _S_ in which to empty WeakRefs, it must empty WeakRefs for all objects in _S_ simultaneously. In other words, an implementation must not empty a WeakRef pointing to an object _obj_ without emptying out other WeakRefs that, if not emptied, could result in an execution that observes the Object value of _obj_.
+        </p>
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-weakref-host-hooks">
+      <h1>Host Hooks</h1>
+
+      <emu-clause id="sec-host-cleanup-finalization-registry" aoid=HostEnqueueFinalizationRegistryCleanupJob>
+        <h1>HostEnqueueFinalizationRegistryCleanupJob ( _finalizationRegistry_ )</h1>
+
+        <p>
+          The abstract operation HostEnqueueFinalizationRegistryCleanupJob takes argument _finalizationRegistry_ (a FinalizationRegistry). HostEnqueueFinalizationRegistryCleanupJob is an implementation-defined abstract operation that is expected to call CleanupFinalizationRegistry(_finalizationRegistry_) at some point in the future, if possible. The host's responsibility is to make this call at a time which does not interrupt synchronous ECMAScript code execution.
+        </p>
+      </emu-clause>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-clear-kept-objects" aoid=ClearKeptObjects>
+    <h1>ClearKeptObjects ( )</h1>
+    <p>
+      The abstract operation CreateRealm takes no arguments. ECMAScript implementations are expected to call ClearKeptObjects when a synchronous sequence of ECMAScript execution completes.
+    </p>
+    <p>The following steps are performed:</p>
+    <emu-alg>
+      1. Let _agent_ be the surrounding agent.
+      1. Set _agent_.[[KeptAlive]] to a new empty List.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-addtokeptobjects" aoid="AddToKeptObjects">
+    <h1>AddToKeptObjects ( _object_ )</h1>
+    <p> The abstract operation AddToKeptObjects takes argument _object_ (an object). The following steps are performed: </p>
+    <emu-alg>
+      1. Let _agent_ be the surrounding agent.
+      1. Append _object_ to _agent_.[[KeptAlive]].
+    </emu-alg>
+    <emu-note>
+      When the abstract operation AddToKeptObjects is called with a target object reference, it adds the target to an identity Set that will point strongly at the target until ClearKeptObjects is called.
```

Comment:
This comment suggests that you want to use a Set for [[KeptAlive]] instead of a List. That sounds fine to me, but isn't what this PR currently does.

Reply:
I have changed it to "a list"